avail/src/main/kotlin/avail/anvil/text/TextKits.kt

```diff
@@ -706,3 +1179,312 @@ internal inline fun JTextPane.transaction(edit: JTextPane.()->Unit)
 		}
 	}
 }
+
+/**
+ * Holds Strings that are queued to be transformed into a different case
+ * representation.
+ *
+ * @author Richard Arriaga
+ *
+ * @constructor
+ * Construct a [StringCaseTransformQueue].
+ *
+ * @param toTransform
+ *   The String be transformed to a different case strategy.
+ */
+internal class StringCaseTransformQueue constructor(toTransform: String)
+{
+	/**
+	 * The Strings that are to be transformed according to the target rules.
+	 */
+	val transformStringQueue: List<String>
+
+	/**
+	 * The non-alphanumeric characters that are placed in between the strings in
+	 * [transformStringQueue].
+	 */
+	val delimiters: List<String>
+
+	init
+	{
+		val tsq = mutableListOf<String>()
+		val d = mutableListOf<String>()
+		var nextDelimiter = ""
+		var nextTransformString = ""
+		var collectingDelimiters = false
+		toTransform.forEach { c ->
+			when
+			{
+				c.isLetterOrDigit() || c == '_' || c == '-' ->
```

Comment:
I hate to do it to you buddy, but you'll need to handle an isolated `-` or `_`, like
```I like - but don't love - traffic lights.```
The (hyphens used as) dashes shouldn't be transformed.  Perhaps this can be handled entirely within `interleave()`, by not transforming the (non-delimiter) strings `"-"` or `"_"`.

Reply:
I think the answer here is to just not transform any string that has less than two characters which can be checked in the `interleave` function:

```kotlin
fun interleave(transformer: (String) -> String): String
{
	val tQueue = transformStringQueue.iterator()
	val dQueue = delimiters.iterator()
	return buildString {
		while (tQueue.hasNext())
		{
			val next = tQueue.next()
			if (next.length < 2)
			{
				append(next)
			}
			else
			{
				append(transformer(next))
			}
			if (dQueue.hasNext())
			{
				append(dQueue.next())
			}
		}
	}
}
```